security_privacy_ln.asciidoc

```diff
@@ -0,0 +1,614 @@
+
+  * Star Heuristic (Funding): if a component contains one source entity that forwards funds to one or more funding entities, these funding entities are likely controlled by the same user.
+  * Snake Heuristic (Funding): if a component contains one source entity that forwards funds to one or more entities, which themselves are used as source and funding entities, then all these entities are likely controlled by the same user.
+  * Collector Heuristic (Settlement): if a component contains one destination entity that receives funds from one or more settlement entities, these settlement entities are likely controlled by the same user.
+  * Proxy Heuristic (Settlement): if a component contains one destination entity that receives funds from one or more entities, which themselves are used as settlement and destination entities, then these entities are likely controlled by the same user.
+
+It is worthwhile pointing out that these heuristics might produce false positives.
+For instance, if transactions of several unrelated users are combined in a CoinJoin transaction, then the Star or the Proxy heuristic can produce false positives.
+This could happen if users are funding a payment channel from a CoinJoin transaction.
+Another potential source of false positives could be that an entity could represent several users if clustered addresses are controlled by a service (e.g., exchange) or on behalf of their users (custodial wallet).
+However, these false positives can effectively be filtered out.
+
+_Countermeasures_: If outputs of funding transactions are not reused for opening other channels, the snake heuristic does not work.
+If users refrain from funding channels from a single external source and avoid collecting funds in a single external destination entity, the other heuristics would not yield any significant results.
+
+*Off-Chain Lightning Node Clustering*
+LN nodes advertise aliases, for instance, _LNBig.com_.
+Aliases can improve the usability of the system.
+However, users tend to use similar aliases for their own different nodes.
+For example, _LNBig.com Billing_ is likely owned by the same user as the node with alias _LNBig.com_.
+Given this observation, one can cluster LN nodes applying their node aliases.
+Specifically, one clusters LN nodes into a single address if their aliases are similar with respect to some string similarity metric.
+
+Another method to cluster LN nodes is applying their IP or Tor addresses.
+If the same IP or Tor addresses correspond to different LN nodes, these nodes are likely controlled by the same user.
+
+_Countermeasures_: For more privacy, aliases should be sufficiently different from one another.
+While the public announcement of IP addresses may be unavoidable for those nodes that wish to have incoming channels in the LN, linkability across nodes of the same user can be mitigated if the clients for each node are hosted with different service providers and thus IP addresses.
```

Comment:
Tor also goes a long way here given that onion services addresses are effectively infinite, while IPv4 adresses are still scarce.